File: FCSeafood/FCSeafood.BusinessObjects/Models/Events/DeliveryModel.cs
namespace FCSeafood.BusinessObjects.Models.Events;

public class DeliveryModel {
    public Guid Id { get; set; }
    public string TrackingNumber { get; set; } = "";
    public UserModel User { get; set; } = new();
    public UserModel? Courier { get; set; }
    public OrderModel Order { get; set; } = new();
    public DeliveryStatusTModel DeliveryStatus { get; set; } = new();
    public PaymentMethodTModel PaymentMethod { get; set; } = new();
    public string? Notes { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.Now;
}

File: FCSeafood-Angular/src/app/common/models/delivery.model.ts
import { UserModel } from "@common-models/user.model";
import { OrderModel } from "@common-models/order.model";
import { DeliveryStatusTModel } from "@common-models/delivery-status-type.model";
import { PaymentMethodTModel } from "@common-models/payment-method-type.model";

export class DeliveryModel {
  id: string = ''; // guid
  trackingNumber: string = '';
  user: UserModel = new UserModel();
  courier: UserModel = new UserModel();
  order: OrderModel = new OrderModel();
  deliveryStatus: DeliveryStatusTModel = new DeliveryStatusTModel();
  paymentMethod: PaymentMethodTModel = new PaymentMethodTModel();
  deliveryFee: number = 0.0;
  notes: string | null = null;
  createdDate: Date = new Date();
}
